pilot/pkg/xds/ads.go

```diff
@@ -453,23 +453,28 @@ func (s *DiscoveryServer) shouldRespond(con *Connection, request *discovery.Disc
 	removed := prev.Difference(cur)
 	added := cur.Difference(prev)
 
-	if len(removed) == 0 && len(added) == 0 {
-		// We should always respond "alwaysRespond" marked requests to let Envoy finish warming
-		// even though Nonce match and it looks like an ACK.
-		if alwaysRespond {
-			log.Infof("ADS:%s: FORCE RESPONSE %s for warming.", stype, con.conID)
-			return true, emptyResourceDelta
-		}
+	// We should always respond "alwaysRespond" marked requests to let Envoy finish warming
```

Comment:
I do not understand why this changed. Do we have a test covering the change?

Reply:
It is not changing the behavior. I think this is added by you, we always respond regardless of sub/unsub.

As you see i move the alwaysRespond in the front.  This is because the change of we donot need to record `Unsubscribed: removed,` now. 

Yes there is a test, if revert this change, there is a test failure.